probe-rs/src/architecture/arm/core/cortex_m.rs

```diff
@@ -80,6 +80,23 @@ impl Mvfr0 {
     }
 }
 
+memory_mapped_bitfield_register! {
+    /// Processor Feature Register 1
+    pub struct IdPfr1(u32);
+    0xE000_ED44, "ID_PFR1",
+    impl From;
+    /// Identifies support for the M-Profile programmer's model
+    pub m_prog_mod, _: 11, 8;
```

Comment:
Mapping this onto an enum with Into/From would be nice and the same for `security` but it's completely optional :)

Reply:
Agreed, I'll make that change.

Reply:
Clarifying the ask here, were you hoping to see something like this?

```rust
memory_mapped_bitfield_register! {
    /// Processor Feature Register 1
    pub struct IdPfr1(u32);
    0xE000_ED44, "ID_PFR1",
    // impl From;
    /// Identifies support for the M-Profile programmer's model
    pub MProgrammersModel, m_prog_mod, _: 11, 8;
    /// Identifies whether the Security Extension is implemented
    pub SecurityExtension, security, _: 7, 4;
}
```

When I tried this, I get a compile error that is beyond my current Rust capability:

```
error[E0277]: the trait bound `u32: BitRange<MProgrammersModel>` is not satisfied
   --> probe-rs/src/architecture/arm/core/cortex_m.rs:117:1
    |
117 | / memory_mapped_bitfield_register! {
118 | |     /// Processor Feature Register 1
119 | |     pub struct IdPfr1(u32);
120 | |     0xE000_ED44, "ID_PFR1",
...   |
125 | |     pub SecurityExtension, security, _: 7, 4;
126 | | }
    | |_^ the trait `BitRange<MProgrammersModel>` is not implemented for `u32`
    |
    = help: the following other types implement trait `BitRange<T>`:
              <u32 as BitRange<i8>>
              <u32 as BitRange<i16>>
              <u32 as BitRange<i32>>
              <u32 as BitRange<i64>>
              <u32 as BitRange<i128>>
              <u32 as BitRange<u8>>
              <u32 as BitRange<u16>>
              <u32 as BitRange<u32>>
            and 2 others
note: required for `IdPfr1` to implement `BitRange<MProgrammersModel>`
   --> probe-rs/src/architecture/arm/core/cortex_m.rs:117:1
    |
117 | / memory_mapped_bitfield_register! {
118 | |     /// Processor Feature Register 1
119 | |     pub struct IdPfr1(u32);
120 | |     0xE000_ED44, "ID_PFR1",
...   |
125 | |     pub SecurityExtension, security, _: 7, 4;
126 | | }
    | |_^ unsatisfied trait bound introduced here
    = note: this error originates in the macro `bitfield_fields` which comes from the expansion of the macro `memory_mapped_bitfield_register` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Any thoughts on how to proceed? Or were you thinking of something else?

Reply:
Alternatively, I can declare those bitfields as `u8` and then use it like this:

```rust
impl IdPfr1 {
    pub fn security_present(&self) -> bool {
        match self.security().into() {
            SecurityExtension::Implemented | SecurityExtension::ImplementedWithStateHandling => {
                true
            }
            _ => false,
        }
    }
}
```

Reply:
The macro can automatically derive the required trait if you provide certain arguments and a From impl. You should be able to see it on other uses of the macro. I am happy to implement it myself at a latter point and merge it as is if you like :)

Reply:
Sounds good. I went ahead and committed the addition of the enums to represent the various bitfield values since I had already defined them and actually found that I was missing a case where `IdPfr1.security_present()` should return true (`SecurityExtension::ImplementedWithStateHandling`).

Reply:
Nice, thanks a lot!